src/language-js/embed/html.js

```diff
@@ -73,17 +79,19 @@ async function embedHtmlLike(parser, textToDoc, print, path, options) {
       ? line
       : null;
 
-  if (linebreak) {
-    return group(["`", indent([linebreak, group(contentDoc)]), linebreak, "`"]);
-  }
-
-  return group([
-    "`",
-    leadingWhitespace,
-    topLevelCount > 1 ? indent(group(contentDoc)) : group(contentDoc),
-    trailingWhitespace,
-    "`",
-  ]);
+  const resultDoc = group(
+    linebreak
+      ? ["`", indent([linebreak, group(contentDoc)]), linebreak, "`"]
+      : [
+          "`",
+          leadingWhitespace,
+          topLevelCount > 1 ? indent(group(contentDoc)) : group(contentDoc),
+          trailingWhitespace,
+          "`",
+        ]
+  );
+
+  return linebreak ? resultDoc : label("[no-hug]", resultDoc);
```

Comment:
`label` seems not limited to `string`, how about

```js
return label({hug: false}, ...)
```

Reply:
Yes, I'll try that.